js/index.js

```diff
@@ -246,30 +244,29 @@ Server.prototype = {
     
     // Get number of plugins that should load
     this.pluginLoader.on('givePluginAmount', event => totalPlugins = event.data);
-    
+
+    let pluginsLoaded = [];
     // Add the plugin and add it to the count
     this.pluginLoader.on('pluginAdded', event => {
-      this.pluginLoaded(event.data);
-      addedPlugins++;            // Make sure total plugins did not get overwritten to null
-
-      if (totalPlugins != null && totalPlugins == addedPlugins) {   // If all the plugins have attempted to load
-        if(eurekaJSON && (eurekaJSON.enabled === true)){
-          var eCopy = JSON.parse(JSON.stringify(eurekaJSON))
-          eCopy.instance = getMediationLayerInstanceObject(webAppOptions, eCopy.instance);
-          eCopy.eureka = getMediationLayerEurekaObject(webAppOptions, eCopy.eureka, eCopy.server);
-          bootstrapLogger.info(`Mediation Layer Configuration = ${JSON.stringify(eCopy,null,2)}`);
-          const client = new eureka(eCopy);
-          delete eCopy.eureka.serviceUrls.default;
-          bootstrapLogger.info('Eureka Instance Id:' + eCopy.instance.instanceId);
-          bootstrapLogger.info("Total Plugins: " + totalPlugins + " Added Plugins: " + addedPlugins);
-          bootstrapLogger.info("############# Connect to Eureka #############");
-          
-          client.logger.level('debug');
-          client.start(function (error) {     // Connect to Eureka Server
-            bootstrapLogger.warn(error || 'complete');
-          });
+      pluginsLoaded.push(event.data);
+      this.pluginLoaded(event.data).then(() => {
+        addedPlugins++;            // Make sure total plugins did not get overwritten to null
+        installLogger.info('Installed plugin: ' + event.data.identifier);
+        if (totalPlugins != null && totalPlugins == addedPlugins) {   // If all the plugins have attempted to load
+          this.finishPluginInitialization(pluginsLoaded, webAppOptions);
         }
-      }
+      }, err => {
+        addedPlugins++;
```

Comment:
"addedPlugins" sounds misleading if it counts plugins that weren't added. Perhaps visitedPlugins?